rllib/env/utils/infinite_lookback_buffer.py

```diff
@@ -397,8 +397,6 @@ def __set(s, n):
 
                 tree.map_structure(__set, self.data, new_data)
             else:
-                if self.space:
```

Comment:
In what cases are changed observations needed? ShouldnÄt all observations come from the same space?

Reply:
Good question. But this check would break any connector pipeline trying to do the following:
* grab an observation (space A) from the episode and put it in the batch
* transform the observation in the batch (basically: perform observation "preprocessing"), now the new observation space is B.
* Insert the transformed observation back into the episode, so we don't have to perform the same transform again in the learner pipeline. Now this insertion will insert a space B observation into a space A episode. That's why - as long a we have NOT finalized the episode and the episode observations are still just a list of (maybe differently shaped) items, we should NOT check the spaces. The connector piece that re-inserts the changed observation back into the episode should then also change the space of the episode from A to B (for those checks that will happen after numpyization/finalization).

Reply:
Okay I understand the logic behind this process. The only thing that could wrong by this later in the process, is that a user checks in the learner conncectors for the space amd does not get what was expected by the space. Changing the space property in the episode or setting a flag (e.g. 'orig_space=False`) or an attribute (e.g. `original_observation_space`) is not possible or just overhead?